app/src/main/java/duke/Task.java

```diff
@@ -0,0 +1,114 @@
+package duke;
+
+public class Task {
+    public static final String ICON_DONE = "X";
+    public static final String ICON_UNDONE = " ";
+
+    protected String details;
+    private boolean isDone;
+
+    /**
+     * Creates a new Task.
+     * @param details What needs to be done.
+     */
+    public Task(String details) {
+        this.details = details;
+        this.isDone = false;
+    }
+
+    /**
+     * Mark a task as done.
+     */
+    public void markDone() {
+        isDone = true;
+    }
+
+    /**
+     * Mark a task as undone.
+     */
+    public void markUndone() {
+        isDone = false;
+    }
+
+    /**
+     * Checks is a task is done.
+     * @return true if the task is done.
+     */
+    public boolean isDone() {
+        return isDone;
+    }
+
+    /**
+     * Creates a Task from an encoded string.
+     * @param s The encoded string.
+     * @return The created task.
+     */
+    public static Task fromEncoded(String s) {
```

Comment:
Interesting way of decoding the input message to decide which task to add. Nice!